### PR TITLE
Keep track of the fhir path while traversing for better errors.

### PR DESCRIPTION
**Pull request by autark**
- Changed the order the FhirConverter is sendt to the conversion routines, so that is is last as it is the least important argument.

### CODE REVIEW

File: src/fhir-tool-conversion/FhirConverter.cs [hunk 1]
```diff
@@ -88,17 +90,26 @@ public static bool CanConvertBetweenVersions(FhirVersion from, FhirVersion to)
...
+                return ConvertObject<TTo, TFrom>(fromObject);
+            }
+            finally
+            {
+                Path = null;
+            }
+        }
 
-             */
+        internal TTo ConvertObject<TTo,TFrom>(TFrom fromObject)
```
Comment (kennethmyhra) [inline src/fhir-tool-conversion/FhirConverter.cs:105]: To be in line with FHIR Terminology this should probably be called ConvertElement.

File: src/fhir-tool-conversion/Converters/Support/FhirR4ToR3ConversionSupport.cs [hunk 2]
```diff
@@ -27,6 +30,16 @@ public override Type GetSourceCodeType()
             return typeof(SourceModel.Code<>);
         }
 
+        public override Type GetTargetFhirElementAttribute()
```
Comment (kennethmyhra) [inline src/fhir-tool-conversion/Converters/Support/FhirR4ToR3ConversionSupport.cs:33]: The name implies that I get the instance of the FhirElementAttribute in return, but I am getting the Type of FhirElementAttribute. Maybe name it GetTargetFhirElementAttributeType and GetSourFhirElementAttributeType

File: src/fhir-tool-conversion/Exceptions/UnknownFhirTypeException.cs [hunk 3]
```diff
@@ -18,6 +18,12 @@ public UnknownFhirTypeException(string message, Type type) : base(message)
             Type = type;
         }
 
+        public UnknownFhirTypeException(Type type, FhirPath path)
```
Comment (kennethmyhra) [inline src/fhir-tool-conversion/Exceptions/UnknownFhirTypeException.cs:21]: Make read-only accessible property for Path

File: src/fhir-tool-conversion/FhirPath.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+using System.Linq;
+using System.Text;
+
+namespace FhirTool.Conversion
+{
+    public class FhirPath
+    {
+        private Stack<string> Path { get; } = new Stack<string>();
+
+        public string GetFullPath()
```
Comment (kennethmyhra) [inline src/fhir-tool-conversion/FhirPath.cs:13]: Add an override for ToString that calls this method.

File: src/fhir-tool-conversion/Exceptions/UnknownFhirTypeException.cs (hunk 3 above)
Comment (autark) [reply src/fhir-tool-conversion/Exceptions/UnknownFhirTypeException.cs:21]: Added accessor for path

File: src/fhir-tool-conversion/FhirPath.cs (hunk 4 above)
Comment (autark) [reply src/fhir-tool-conversion/FhirPath.cs:13]: Added override for ToString() in FhirPath

**Pull request merged**